script/test/cmd/cmd_test.go

```diff
@@ -0,0 +1,775 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"github.com/joho/godotenv"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"reflect"
+	"regexp"
+	"strings"
+	"testing"
+)
+
+func fileNotExist(path string) bool {
+	return findExecutable(path) != nil
+}
+func findExecutable(file string) error {
+	d, err := os.Stat(file)
+	if err != nil {
+		return err
+	}
+	if m := d.Mode(); !m.IsDir() {
+		return nil
+	}
+	return os.ErrPermission
+}
+
+func binary(cmd string) string {
+	path, err := exec.LookPath(cmd)
+	if err != nil || fileNotExist(cmd) {
+		home := os.Getenv("HOME")
+		if home == "" {
+			fmt.Println("No $HOME environment variable found")
+		}
+	}
+	return path
+}
+
+//Git get the branch
+func Git() (branch string, error error) {
+	cmdName := "git"
+	cmdArgs := []string{"rev-parse", "--abbrev-ref", "HEAD"}
+	cmdOut, err := exec.Command(cmdName, cmdArgs...).CombinedOutput()
+	if err != nil {
+		fmt.Println("")
+		return "", errors.Wrap(err, "Error getting branch name")
+	}
+	branch = string(cmdOut)
+	return branch, nil
+}
+
+//UpdateUri Replaces variables with current branch and uri
+func UpdateURI(uri, branch, inputfile, output string) error {
+	input, _ := ioutil.ReadFile(inputfile)
+	file := string(input)
+	r, _ := regexp.Compile("%URI%")
+	f := r.ReplaceAllString(file, uri)
+	k, _ := regexp.Compile("%REF%")
+	g := k.ReplaceAllString(f, branch)
+	err := ioutil.WriteFile(output, []byte(g), 0644)
+	if err != nil {
+		return errors.Wrapf(err, "Error updating the file with %s and %s", uri, branch)
+	}
+	return nil
+}
+
+//Get origin
+func GitOrigin() (origin, branch string, error error) {
+	cmdName := "git"
+	cmdArgs := []string{"remote", "get-url", "origin"}
+	cmdOut, err := exec.Command(cmdName, cmdArgs...).CombinedOutput()
+	if err != nil {
+		fmt.Println("")
+		return "", "", errors.Wrap(err, "Error getting origin.")
+	}
+	origin = string(cmdOut)
+	origin = strings.TrimSpace(origin)
+	if !strings.Contains(origin, ".git") {
+		origin = origin + ".git"
+	}
+	brch, err := Git()
+	if err != nil {
+		log.Printf("%+v", err)
+		return
+	}
+	branch = strings.TrimSpace(brch)
+	fmt.Println("Test regarding build context running kompose from various directories")
+	fmt.Println("Buildconfig using " + origin + "::" + branch + " as source.")
+	return origin, branch, nil
+}
+
+// Kompose command for buildContext
+func KomposeCommand(command, provider, outFile, dockerComposeFile string) (error error) {
+	file := strings.Split(dockerComposeFile, "/")
+	cmdName := "kompose"
+	cmdArgs := []string{command, "--provider", provider, "-f", dockerComposeFile, "--stdout", "-j"}
+	out, err := exec.Command(cmdName, cmdArgs...).Output()
+	if err != nil {
+		return errors.Wrapf(err, "Error in %s application", file[1])
+	}
+	er := ioutil.WriteFile(outFile, out, 0644)
```

Comment:
Let's do `err` here, and reuse the `err` variable already created without using `:=`